src/Compilers/CSharp/CSharpAnalyzerDriver/CSharpDeclarationComputer.cs

```diff
@@ -231,6 +246,35 @@ private static bool InvalidLevel(int? level)
             }
         }
 
+        private static IEnumerable<SyntaxNode> GetAttributes(TypeDeclarationSyntax typeDeclaration)
+        {
+            switch (typeDeclaration.Kind())
```

Comment:
>switch (typeDeclaration.Kind()) [](start = 12, length = 31)

Is the switch necessary? It looks like TypeDeclarationSyntax has AttributeLists property.

Reply:
Perhaps this overload isn't even needed. Is it used only in one place?

---
In reply to: [152410584](https://github.com/dotnet/roslyn/pull/23318#discussion_r152410584) [](ancestors = 152410584)